pinot-core/src/main/java/org/apache/pinot/core/query/pruner/SegmentPrunerService.java

```diff
@@ -38,11 +39,21 @@
 
   public SegmentPrunerService(SegmentPrunerConfig config) {
     int numPruners = config.numSegmentPruners();
-    _segmentPruners = new ArrayList<>(numPruners);
+    _segmentPruners = new ArrayList<>(numPruners + 1);
```

Comment:
We should always add `ValidSegmentPruner` as the first pruner?

Reply:
yeah, changed it to be that way